pkg/asset/ignition/bootstrap/bootstrap_ignition.go

```diff
@@ -36,6 +38,20 @@ func parseCertificateBundle(userCA []byte) ([]igntypes.Resource, error) {
 // GenerateIgnitionShimWithCertBundle is used to generate an ignition file that contains a user ca bundle
 // in its Security section.
 func GenerateIgnitionShimWithCertBundle(bootstrapConfigURL string, userCA string) ([]byte, error) {
+	ign, err := generateIgnitionShimWithCertBundle(bootstrapConfigURL, userCA)
```

Comment:
Trying to figure out why we'd want to keep this non-proxy form around:

```console
$ git --no-pager log --oneline -1 origin/pr/5973
0defd86d01 (origin/pr/5973) Bootstrap: AWS - GenerateIgnitionShimWithCertBundleAndProxy should reuse function GenerateIgnitionShimWithCertBundle
$ git --no-pager grep 'GenerateIgnitionShimWithCertBundle(' origin/pr/5973
origin/pr/5973:pkg/asset/cluster/tfvars.go:                     shim, err := bootstrap.GenerateIgnitionShimWithCertBundle(bootstrapIgnURLPlaceholder, installConfig.Config.AdditionalTrustBundle)
origin/pr/5973:pkg/asset/ignition/bootstrap/bootstrap_ignition.go:func GenerateIgnitionShimWithCertBundle(bootstrapConfigURL string, userCA string) ([]byte, error) {
origin/pr/5973:pkg/tfvars/alibabacloud/alibabacloud.go: stubIgn, err := bootstrap.GenerateIgnitionShimWithCertBundle(sources.IgnitionPresignedURL, sources.AdditionalTrustBundle)
```

That's some `AzureStackCloud` thing, and an Alibaba thing.  Presumably they'd both want the proxy configuration too?  But I'm not an approver here, and it's fine if folks want to handle the providers one at a time, instead of all in a single pull request.

Reply:
Hi @wking 
I agree, especially that, looking at the documentation, we don't seem to distinguish Proxy Setup for each cloud provider...
I'm happy to extend this if we all agree on this.
Openstack implementation has its own `bootstrap_ignition.go` and its own `GenerateIgnitionShimWithCertBundleAndProxy`, so we could also reuse the code there (requires some changes).